tests/rptest/transactions/compaction_e2e_test.py

```diff
@@ -29,7 +29,7 @@ class CompactionE2EIdempotencyTest(RedpandaTest):
     def __init__(self, test_context):
         extra_rp_conf = {}
 
-        self.segment_size = 5 * 1024 * 1024
+        self.segment_size = 1 * 1024 * 1024
```

Comment:
curious why?

Reply:
this test was failing to me even before the refactoring.